### ISSUE TITLE
[help wanted]Issue with KZ

### ISSUE THREAD

**Issue by 0uck1** [kus/csgo-modded-server#12]
Hi,
i have installed your rep on my rented server and for the most part it is working rly fine.
Nevertheless i got problems if i switch the configuration to Kreedz Climbing.
If i switch, i instantly lose the connection to the server and i cant reconnect anymore. 

If i try to reconnect, i get the following output in the server console:

```
Mar 14 16:00:34:  -> Reservation cookie dbe347f7f0ee4c58:  reason [R] Connect from *******
Mar 14 16:00:34:  ---- Host_Changelevel ----
Mar 14 16:00:35:  *** Map Load: kz_breezeblocks: Map Group mg_activeGameTypes: missing mapgroupsSP entry for game type/mode (custom/custom).
Mar 14 16:00:35:  GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/cooperative).
Mar 14 16:00:35:  GameTypes: missing mapgroupsSP entry for game type/mode (cooperative/coopmission).
Mar 14 16:00:35:  ConVarRef room_type doesn't point to an existing ConVar
Mar 14 16:00:35:  mp_solid_teammates - 1
Mar 14 16:00:35:  mp_respawn_on_death_ct - 0
Mar 14 16:00:35:  mp_respawn_on_death_t - 0
Mar 14 16:00:35:  mp_ignore_round_win_conditions - 0
Mar 14 16:00:35:  mp_teammates_are_enemies - 0
Mar 14 16:00:35:  mp_death_drop_gun - 1
Mar 14 16:00:35:  mp_respawn_immunitytime - 4.0
Mar 14 16:00:35:  mp_respawn_on_death - missing cvar specified in bspconvar_whitelist.txt
Mar 14 16:00:35:  mp_weapons_allow_map_placed - 0
Mar 14 16:00:35:  Executing dedicated server config file
Mar 14 16:00:35:  [SM] Changed cvar "phys_pushscale" to "1".
Mar 14 16:00:35:  [SM] Changed cvar "phys_pushscale" to "1".
Mar 14 16:00:35:  [SM] Changed cvar "phys_timescale" to "1".
Mar 14 16:00:35:  [SM] Changed cvar "sv_client_min_interp_ratio" to "1".
Mar 14 16:00:35:  [SM] Changed cvar "sv_client_max_interp_ratio" to "5".
Mar 14 16:00:35:  Writing cfg/banned_user.cfg.
Mar 14 16:00:35:  Writing cfg/banned_ip.cfg.
Mar 14 16:00:35:  Warning: Game server already logged into steam.  You need to use the sv_setsteamaccount command earlier.
Mar 14 16:00:35:  env.cfg executed 
Mar 14 16:00:35:  server.cfg executed 
Mar 14 16:00:35:  exec: couldn't exec gamemode_casual_server.cfg
Mar 14 16:00:35:  CHostage::Precache: missing hostage models for map kz_breezeblocks. Adding the default models.
Mar 14 16:00:36:  Commentary: Could not find commentary data file 'maps/kz_breezeblocks_commentary.txt'. 
Mar 14 16:00:36:  Error parsing BotProfile.db - unknown attribute 'Rank'
Mar 14 16:00:36:  message repeated 7 times: [ Error parsing BotProfile.db - unknown attribute 'Rank']
```

And the following in my client console:
```
**** Unable to localize '#matchdraft_phase_action_wait' on panel 'id-map-draft-phase-wait'
**** Unable to localize '#DemoPlayback_Restart' on panel descendant of 'HudDemoPlayback'
**** Unable to localize '#DemoPlayback_Back' on panel descendant of 'HudDemoPlayback'
**** Unable to localize '#DemoPlayback_Pause' on panel descendant of 'HudDemoPlayback'
**** Unable to localize '#DemoPlayback_Slow' on panel descendant of 'HudDemoPlayback'
**** Unable to localize '#DemoPlayback_Play' on panel descendant of 'HudDemoPlayback'
**** Unable to localize '#DemoPlayback_Fast' on panel descendant of 'HudDemoPlayback'
**** Unable to localize '#DemoPlayback_Next' on panel descendant of 'HudDemoPlayback'
**** Unable to localize '#Panorama_CSGO_Spray_Cursor_Hint' on panel 'RosettaInfoText'
ChangeGameUIState: CSGO_GAME_UI_STATE_MAINMENU -> CSGO_GAME_UI_STATE_LOADINGSCREEN
Connecting to public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Server using 'friends' lobbies, requiring pw yes, lobby id ffffffffffffffff
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Retrying public(51.195.60.183:27949) ...
Connection failed after 30 retries.

**** Unable to localize '#GenericConfirmText_Label' on panel descendant of 'PopupManager'
ChangeGameUIState: CSGO_GAME_UI_STATE_LOADINGSCREEN -> CSGO_GAME_UI_STATE_MAINMENU
ChangeGameUIState: CSGO_GAME_UI_STATE_MAINMENU -> CSGO_GAME_UI_STATE_MAINMENU
Resource materials\panorama\images\icons\ui\globe.svg failed to load.
PNG load error Interlace handling should be turned on when using png_read_image
```

Do you have an idea what i am doing wrong or how i can fix that?

Other than that, thank you for this awesome collection!!!

**Comment by kus** [comment]
Hi @0uck1

Just an update, I've been trying to rectify this, tried different versions of KZTimer, disabling some extensions but can't get it to work.

The weird thing is it used to work on my server but I haven't personally changed to the mod in about a year.

I've lodged a support ticket with KZTimer team with lots of debug information and error logs to try get it sorted.

Will post here when it is.

Thanks for bringing to my attention.

For my reference, keeping debug information here:

I've started getting this error on my server when trying to turn on KZTimer:

```L 03/18/2022 - 03:16:26: [SM] Exception reported: Invalid database Handle 0 (error: 4)
L 03/18/2022 - 03:16:26: [SM] Blaming: disabled/KZTimerGlobal.smx
L 03/18/2022 - 03:16:26: [SM] Call stack trace:
L 03/18/2022 - 03:16:26: [SM]   [0] SQL_TQuery
L 03/18/2022 - 03:16:26: [SM]   [1] Line 5055, kztimerGlobal/sql.sp::db_viewPlayerOptions
L 03/18/2022 - 03:16:26: [SM]   [2] Line 1319, E:\git\kztimerglobal\scripting\KZTimerGlobal.sp::OnClientPutInSe$
L 03/18/2022 - 03:16:26: [SM]   [3] Line 115, kztimerGlobal/triggerfix.sp::OnPluginStart_Triggerfix
L 03/18/2022 - 03:16:26: [SM]   [4] Line 874, E:\git\kztimerglobal\scripting\KZTimerGlobal.sp::OnPluginStart```

From what I can see, sqlite is working for other plugins.

Contents of `/home/steam/csgo/csgo/addons/sourcemod/data/sqlite`

```-rw-r--r-- 1 steam steam  37K Mar 18 03:13 clientprefs-sqlite.sq3
-rw-r--r-- 1 steam steam  51K Jun  2  2021 kztimer-sqlite.sq3
-rw-r--r-- 1 steam steam  24K Mar 14 11:29 rankme.sq3
-rw-r--r-- 1 steam steam  68K Mar 18 03:16 sourcemod-local.sq3```

I can open `kztimer-sqlite.sq3` via terminal and list the tables so the file doesn't appear to be corrupt.

It used to work with no issues, but I haven't changed to the KZ mod in a long time on my server, and now it doesn't work.

I run this github repo for the community where I have a server with multiple mods that you can switch between https://github.com/kus/csgo-modded-server and I keep them up-to-date.

Defaultly no mods are loaded, the plugins are in a disabled folder and manually loaded when you exec the config.

I updated to version `1.100` which still didn't work and tried `1.99.2` which also doesn't work.

Here is the `database.cfg` https://github.com/kus/csgo-modded-server/blob/master/csgo/addons/sourcemod/configs/databases.cfg

Here is what is executed when you want to switch to KZ mod https://github.com/kus/csgo-modded-server/blob/master/csgo/cfg/kz.cfg

Here is the logs of when I load the kz mod:

```L 03/18/2022 - 03:16:25: [basecommands.smx] "Kus<12><STEAM_1:0:56050><>" executed config (file "kz.cfg")
L 03/18/2022 - 03:16:25: [basecommands.smx] "Console<0><Console><Console>" changed cvar (cvar "mapcyclefile") ($
L 03/18/2022 - 03:16:26: [GEOIP] GeoIP database info: GEO-106FREE 20180327 Build 1 Copyright (c) 2018 MaxMind I$
L 03/18/2022 - 03:16:26: [FutureExe.smx] Console<0><Console><Console> used Future-Execute to execute command ch$
L 03/18/2022 - 03:16:26: [weapons.smx] SQLite DB connection successful
L 03/18/2022 - 03:16:26: [weapons.smx] Inactive players' data has been deleted
L 03/18/2022 - 03:16:29: [FutureExe.smx] "CONSOLE"(CONSOLE) used Future-Execute to execute command changelevel $
L 03/18/2022 - 03:16:30: -------- Mapchange to kz_breezeblocks --------
L 03/18/2022 - 03:16:30: [basecommands.smx] "Console<0><Console><Console>" changed cvar (cvar "phys_pushscale")$
L 03/18/2022 - 03:16:30: [basecommands.smx] "Console<0><Console><Console>" changed cvar (cvar "phys_pushscale")$
L 03/18/2022 - 03:16:30: [basecommands.smx] "Console<0><Console><Console>" changed cvar (cvar "phys_timescale")$
L 03/18/2022 - 03:16:30: [basecommands.smx] "Console<0><Console><Console>" changed cvar (cvar "sv_client_min_in$
L 03/18/2022 - 03:16:30: [basecommands.smx] "Console<0><Console><Console>" changed cvar (cvar "sv_client_max_in$
L 03/18/2022 - 03:16:33: [mapconfigs.smx] Searching specific configs for kz_breezeblocks
L 03/18/2022 - 03:16:33: [mapconfigs.smx] Executing map specific config: kz_.cfg
L 03/18/2022 - 03:16:33: [weapons.smx] Inactive players' data has been deleted```

Plugins:

```[SM] Listing 25 plugins:
  01 "Client Preferences" (1.10.0.6499) by AlliedModders LLC
  02 "Player Commands" (1.10.0.6499) by AlliedModders LLC
  03 "Admin Menu" (1.10.0.6499) by AlliedModders LLC
  04 "Basic Ban Commands" (1.10.0.6499) by AlliedModders LLC
  05 "Fun Votes" (1.10.0.6499) by AlliedModders LLC
  06 "Sound Commands" (1.10.0.6499) by AlliedModders LLC
  07 "Command Time-Traveler" (1.2.0) by DarthNinja
  08 "Anti-Flood" (1.10.0.6499) by AlliedModders LLC
  09 "Nextmap" (1.10.0.6499) by AlliedModders LLC
  10 "Fun Commands" (1.10.0.6499) by AlliedModders LLC
  11 "Fix Hint Color Messages" (1.2.2 Franc1sco franug github version) by Phoenix (˙·٠●Феникс●٠·˙)
  12 "Reserved Slots" (1.10.0.6499) by AlliedModders LLC
  13 "Gloves" (1.0.5) by kgns | oyunhost.net
  14 "Basic Comm Control" (1.10.0.6499) by AlliedModders LLC
  15 "Map configs" (1.3) by Berni
  16 "Basic Votes" (1.10.0.6499) by AlliedModders LLC
  17 "Basic Commands" (1.10.0.6499) by AlliedModders LLC
  18 "Basic Info Triggers" (1.10.0.6499) by AlliedModders LLC
  19 "Admin File Reader" (1.10.0.6499) by AlliedModders LLC
  20 "Basic Chat" (1.10.0.6499) by AlliedModders LLC
  21 "Weapons & Knives" (1.7.1) by kgns | oyunhost.net
  22 "Admin Help" (1.10.0.6499) by AlliedModders LLC
  23 "MapChooser" (1.10.0.6499) by AlliedModders LLC
  24 "NoBlock" (1.4.2) by Otstrel.ru Team
  25 "KZErrorBoxFixer" (1.1) by 1NutWunDeR```

  I read somewhere that PTaH can cause an issue with KZTimer, I removed it and still have the same issue.

**Comment by kus** [comment]
@0uck1 after doing a lot of testing, it seems like KZTimer can't be hot loaded anymore like every other mod in my collection.

So the solution is to have it always load with the server, and based on the map type to unload.

This does cause a weird problem that when you first boot the server the map keeps ending/changing. So you need to either change the server to the mod you want to play GunGame, Competitive, Wingman, KZ etc and it's fixed.

You can do this via the admin menu, RCON or from the server console i.e. `exec gg`.

If you want to play KZ, you must do it as the first mod. You can't do it after another mod. So when the server starts up, via admin menu, RCON or server console run `exec kz`.

If you want to play KZ AFTER having another mod already loaded (i.e. GunGame) you need to restart the server and then load it, you can do this by running the command `exec sourcemod/restart` via the admin menu (Server EXEC > Restart), RCON or from the server console.

Please let me know if you have any issues.
